### ISSUE TITLE
better ARIA for button groups within toolbars

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#6522]
Building on #6521, often times the view-switching buttons are rendered as a "button-group". It would be nice to semantically define these as a group so that screen readers can announce the top-level widget instead of announcing all child buttons.

We can automatically detect adjacent view-switching buttons. If there are more than one, we group them under a role="group". [Bootstrap shows a nice example of this](https://getbootstrap.com/docs/5.1/components/button-group/#basic-example).

The aria-label will be a static property within the locale files:

```js
buttonAriaLabels: {
  viewGroup: 'Change view'
}
```

This will be a breaking change because it changes the DOM structure.

**Comment by laryn** [comment]
This may be related. I'm testing this calendar on a dev site and an internal accessibility review came back with the following:

> Interacting with the month/week, and the arrows does not announce to a user of changes: aria-live should probably be utilized.

**Comment by arshaw** [comment]
In [v7.0.0-beta.4](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.4), consecutive view buttons are inferred as a tablist. Demo: https://codepen.io/arshaw/pen/QwLQMyX?editors=0010

**Comment by clareaokeeffe** [comment]
> In [v7.0.0-beta.4](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.4), consecutive view buttons are inferred as a tablist. Demo: https://codepen.io/arshaw/pen/QwLQMyX?editors=0010

If you're going to use an ARIA tablist then there need to be corresponding elements with role="tabpanel" with their visibility being toggled in response to the tab selection. These are missing in the linked demo. Also, in the demo the [expected keyboard interaction](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#keyboardinteraction) of arrowing between tabs (not TABing) has not been implemented.

Ultimately, the previous state of using buttons with aria-pressed is perfectly functional. Some a11y tester just saw something that looks kinda like tabs and therefore said "it must be tabs!" without giving it enough thought. 

However, adding role="group" and an aria-label on the container around the buttons (as originally mentioned) would be good.
